keanu-python/tests/test_net.py

```diff
@@ -62,3 +63,32 @@ def test_probe_for_non_zero_probability_from_bayes_net() -> None:
 
     assert gamma.has_value()
     assert poisson.has_value()
+
+
+def test_can_save_and_load() -> None:
+    PROTO_FILE = "temp.proto"
+    JSON_FILE = "temp.json"
+    DOT_FILE = "temp.dot"
+    gamma = Gamma(1.0, 1.0)
+    gamma.set_value(2.5)
+    net = BayesNet(gamma.get_connected_graph())
+    metadata = {"Team": "GraphOS"}
+    protobuf_saver = ProtobufSaver(net)
+    protobuf_saver.save(PROTO_FILE, True, metadata)
+    json_saver = JsonSaver(net)
+    json_saver.save(JSON_FILE, True, metadata)
+    dot_saver = DotSaver(net)
+    dot_saver.save(DOT_FILE, True, metadata)
```

Comment:
Can you assert something about the `DOT_FILE`? Ideally, diff it against another file that you expect to be identical.

Reply:
I've avoided doing that given there's already substantial testing on the Java side and don't want to add unnecessary testing.  Would checking that there are the correct number of lines in the file be sufficient?

Reply:
That's a fair point.

You could do a bit more than counting lines, e.g. they always start with "`digraph BayesianNetwork {`"